### DISCUSSION TITLE
How do I batch multiple useSWR calls?

### DISCUSSION THREAD

**Discussion by renatoi** [vercel/swr#1249] [Q&A]
I need to be able to do a bunch of useSWR calls from a list of ids (size is unknown and can change).

```js
const ids = ['123', '456', '789', /* ... there could be many more */];

useSWR(`api/{ids[0]}`, fetcher)
useSWR(`api/{ids[1]}`, fetcher)
useSWR(`api/{ids[2]}`, fetcher)
// ...
```

I can't put them in a for loop because it's a hook, and I also want to know when all are finished, similar to `Promise.all()`.

I'm confused by `useSWR` in this situation because it's a hook and it doesn't look like it returns a promise so how do I batch all these calls? What's the best way to do this?

**Comment by promer94** [comment]
#1041 might be able to give you some hint
